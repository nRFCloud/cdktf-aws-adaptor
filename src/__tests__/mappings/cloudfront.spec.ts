import { CloudfrontDistribution } from "@cdktf/provider-aws/lib/cloudfront-distribution/index.js";
import { CloudfrontOriginAccessIdentity } from "@cdktf/provider-aws/lib/cloudfront-origin-access-identity/index.js";
import { CfnCloudFrontOriginAccessIdentity, CfnDistribution } from "aws-cdk-lib/aws-cloudfront";
import { itShouldMapCfnElementToTerraformResource } from "../helpers.js";

describe("CloudFront", () => {
  itShouldMapCfnElementToTerraformResource(
    CfnCloudFrontOriginAccessIdentity,
    {
      cloudFrontOriginAccessIdentityConfig: {
        comment: "test",
      },
    },
    CloudfrontOriginAccessIdentity,
    {
      comment: "test",
    },
  );

  itShouldMapCfnElementToTerraformResource(
    CfnDistribution,
    {
      tags: [{
        key: "Name",
        value: "my-distribution",
      }],
      distributionConfig: {
        defaultCacheBehavior: {
          allowedMethods: ["GET", "HEAD"],
          cachedMethods: ["GET", "HEAD"],
          compress: true,
          defaultTtl: 3600,
          forwardedValues: {
            cookies: {
              forward: "none",
            },
            queryString: false,
            headers: ["*"],
            queryStringCacheKeys: ["*"],
          },
          cachePolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
          lambdaFunctionAssociations: [
            {
              eventType: "origin-request",
              lambdaFunctionArn: "arn:aws:lambda:us-east-1:111111111111:function:my-function",
              includeBody: true,
            },
          ],
          fieldLevelEncryptionId: "1234567890abcdef1234567890abcdef",
          maxTtl: 86400,
          minTtl: 0,
          originRequestPolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
          realtimeLogConfigArn: "arn:aws:logs:us-east-1:111111111111:realtime-log-config/my-distribution",
          responseHeadersPolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
          smoothStreaming: true,
          targetOriginId: "my-target-origin",
          trustedKeyGroups: ["1234567890abcdef1234567890abcdef"],
          functionAssociations: [
            {
              eventType: "viewer-request",
              functionArn: "arn:aws:lambda:us-east-1:111111111111:function:my-function",
            },
          ],
          trustedSigners: ["1234567890abcdef1234567890abcdef"],
          viewerProtocolPolicy: "redirect-to-https",
        },
        aliases: ["example.com"],
        comment: "My CloudFront distribution",
        customErrorResponses: [
          {
            errorCode: 404,
            responseCode: 200,
            responsePagePath: "/index.html",
            errorCachingMinTtl: 300,
          },
        ],
        defaultRootObject: "index.html",
        enabled: true,
        httpVersion: "http2",
        ipv6Enabled: true,
        priceClass: "PriceClass_100",
        restrictions: {
          geoRestriction: {
            restrictionType: "whitelist",
            locations: ["US", "CA", "GB"],
          },
        },
        viewerCertificate: {
          acmCertificateArn: "arn:aws:acm:us-east-1:111111111111:certificate/12345678-1234-1234-1234-123456789012",
          minimumProtocolVersion: "TLSv1.2_2019",
          sslSupportMethod: "sni-only",
        },
        cacheBehaviors: [
          {
            allowedMethods: ["GET", "HEAD"],
            cachedMethods: ["GET", "HEAD"],
            compress: true,
            defaultTtl: 3600,
            forwardedValues: {
              cookies: {
                forward: "none",
              },
              queryString: false,
              headers: ["*"],
              queryStringCacheKeys: ["*"],
            },
            cachePolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
            lambdaFunctionAssociations: [
              {
                eventType: "origin-request",
                includeBody: true,
                lambdaFunctionArn: "arn:aws:lambda:us-east-1:111111111111:function:my-function",
              },
            ],
            fieldLevelEncryptionId: "1234567890abcdef1234567890abcdef",
            maxTtl: 86400,
            minTtl: 0,
            originRequestPolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
            pathPattern: "/images/*",
            realtimeLogConfigArn: "arn:aws:logs:us-east-1:111111111111:realtime-log-config/my-distribution",
            responseHeadersPolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
            smoothStreaming: true,
            targetOriginId: "my-target-origin",
            viewerProtocolPolicy: "redirect-to-https",
            trustedKeyGroups: ["1234567890abcdef1234567890abcdef"],
            trustedSigners: ["1234567890abcdef1234567890abcdef"],
            functionAssociations: [
              {
                eventType: "viewer-request",
                functionArn: "arn:aws:lambda:us-east-1:111111111111:function:my-function",
              },
            ],
          },
        ],
        cnamEs: ["example.com"],
        continuousDeploymentPolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
        logging: {
          bucket: "my-logs.s3.amazonaws.com",
          includeCookies: true,
          prefix: "myprefix",
        },
        originGroups: {
          items: [
            {
              members: {
                items: [
                  {
                    originId: "my-origin",
                  },
                ],
                quantity: 1,
              },
              id: "my-origin-group",
              failoverCriteria: {
                statusCodes: {
                  items: [500, 502],
                  quantity: 2,
                },
              },
            },
          ],
          quantity: 1,
        },
        webAclId: "arn:aws:wafv2:us-east-1:111111111111:regional/webacl/my-web-acl",
        origins: [{
          domainName: "example.com",
          id: "my-origin",
          connectionAttempts: 3,
          connectionTimeout: 10,
          customOriginConfig: {
            httpPort: 80,
            httpsPort: 443,
            originKeepaliveTimeout: 5,
            originProtocolPolicy: "https-only",
            originReadTimeout: 30,
            originSslProtocols: ["TLSv1.2"],
          },
          originAccessControlId: "1234567890abcdef1234567890abcdef",
          originPath: "/mypath",
          originShield: {
            enabled: true,
          },
          originCustomHeaders: [{
            headerName: "MyCustomHeader",
            headerValue: "MyCustomValue",
          }],
          s3OriginConfig: {
            originAccessIdentity: "origin-access-identity/cloudfront/E127EXAMPLE51Z",
          },
        }],
        staging: true,
      },
    },
    CloudfrontDistribution,
    {
      tags: {
        Name: "my-distribution",
      },
      defaultCacheBehavior: {
        allowedMethods: ["GET", "HEAD"],
        cachedMethods: ["GET", "HEAD"],
        compress: true,
        defaultTtl: 3600,
        forwardedValues: {
          cookies: {
            forward: "none",
          },
          queryString: false,
          headers: ["*"],
          queryStringCacheKeys: ["*"],
        },
        cachePolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
        lambdaFunctionAssociation: [
          {
            eventType: "origin-request",
            lambdaArn: "arn:aws:lambda:us-east-1:111111111111:function:my-function",
            includeBody: true,
          },
        ],
        fieldLevelEncryptionId: "1234567890abcdef1234567890abcdef",
        maxTtl: 86400,
        minTtl: 0,
        originRequestPolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
        realtimeLogConfigArn: "arn:aws:logs:us-east-1:111111111111:realtime-log-config/my-distribution",
        responseHeadersPolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
        smoothStreaming: true,
        targetOriginId: "my-target-origin",
        trustedKeyGroups: ["1234567890abcdef1234567890abcdef"],
        functionAssociation: [
          {
            eventType: "viewer-request",
            functionArn: "arn:aws:lambda:us-east-1:111111111111:function:my-function",
          },
        ],
        trustedSigners: ["1234567890abcdef1234567890abcdef"],
        viewerProtocolPolicy: "redirect-to-https",
      },
      aliases: ["example.com", "example.com"],
      comment: "My CloudFront distribution",
      customErrorResponse: [
        {
          errorCode: 404,
          responseCode: 200,
          responsePagePath: "/index.html",
          errorCachingMinTtl: 300,
        },
      ],
      defaultRootObject: "index.html",
      enabled: true,
      httpVersion: "http2",
      isIpv6Enabled: true,
      priceClass: "PriceClass_100",
      restrictions: {
        geoRestriction: {
          restrictionType: "whitelist",
          locations: ["US", "CA", "GB"],
        },
      },
      viewerCertificate: {
        acmCertificateArn: "arn:aws:acm:us-east-1:111111111111:certificate/12345678-1234-1234-1234-123456789012",
        minimumProtocolVersion: "TLSv1.2_2019",
        sslSupportMethod: "sni-only",
      },
      orderedCacheBehavior: [
        {
          allowedMethods: ["GET", "HEAD"],
          cachedMethods: ["GET", "HEAD"],
          compress: true,
          defaultTtl: 3600,
          forwardedValues: {
            cookies: {
              forward: "none",
            },
            queryString: false,
            headers: ["*"],
            queryStringCacheKeys: ["*"],
          },
          cachePolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
          lambdaFunctionAssociation: [
            {
              eventType: "origin-request",
              includeBody: true,
              lambdaArn: "arn:aws:lambda:us-east-1:111111111111:function:my-function",
            },
          ],
          fieldLevelEncryptionId: "1234567890abcdef1234567890abcdef",
          maxTtl: 86400,
          minTtl: 0,
          originRequestPolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
          pathPattern: "/images/*",
          realtimeLogConfigArn: "arn:aws:logs:us-east-1:111111111111:realtime-log-config/my-distribution",
          responseHeadersPolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
          smoothStreaming: true,
          targetOriginId: "my-target-origin",
          viewerProtocolPolicy: "redirect-to-https",
          trustedKeyGroups: ["1234567890abcdef1234567890abcdef"],
          trustedSigners: ["1234567890abcdef1234567890abcdef"],
          functionAssociation: [
            {
              eventType: "viewer-request",
              functionArn: "arn:aws:lambda:us-east-1:111111111111:function:my-function",
            },
          ],
        },
      ],
      continuousDeploymentPolicyId: "658327ea-f89d-4fab-a63d-7e88639e58f6",
      loggingConfig: {
        bucket: "my-logs.s3.amazonaws.com",
        includeCookies: true,
        prefix: "myprefix",
      },
      originGroup: [
        {
          failoverCriteria: {
            statusCodes: [500, 502],
          },
          member: [{
            originId: "my-origin",
          }],
          originId: "my-origin-group",
        },
      ],
      webAclId: "arn:aws:wafv2:us-east-1:111111111111:regional/webacl/my-web-acl",
      origin: [
        {
          connectionAttempts: 3,
          connectionTimeout: 10,
          customHeader: [
            {
              name: "MyCustomHeader",
              value: "MyCustomValue",
            },
          ],
          customOriginConfig: {
            httpPort: 80,
            httpsPort: 443,
            originKeepaliveTimeout: 5,
            originProtocolPolicy: "https-only",
            originReadTimeout: 30,
            originSslProtocols: ["TLSv1.2"],
          },
          domainName: "example.com",
          originAccessControlId: "1234567890abcdef1234567890abcdef",
          originId: "my-origin",
          originPath: "/mypath",
          originShield: {
            enabled: true,
          },
          s3OriginConfig: {
            originAccessIdentity: "origin-access-identity/cloudfront/E127EXAMPLE51Z",
          },
        },
      ],
      staging: true,
    },
  );
});
