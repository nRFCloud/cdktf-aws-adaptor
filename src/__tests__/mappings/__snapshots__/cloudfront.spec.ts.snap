// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CloudFront > Should map AWS::CloudFront::CloudFrontOriginAccessIdentity 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cloudfront_origin_access_identity\\": {
      \\"resource_22C949BF\\": {
        \\"comment\\": \\"test\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_cloudfront_origin_access_identity.resource_22C949BF.id, \\\\\\",\\\\\\", aws_cloudfront_origin_access_identity.resource_22C949BF.s3_canonical_user_id])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`CloudFront > Should map AWS::CloudFront::Distribution 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cloudfront_distribution\\": {
      \\"resource_22C949BF\\": {
        \\"aliases\\": [
          \\"example.com\\",
          \\"example.com\\"
        ],
        \\"comment\\": \\"My CloudFront distribution\\",
        \\"continuous_deployment_policy_id\\": \\"658327ea-f89d-4fab-a63d-7e88639e58f6\\",
        \\"custom_error_response\\": [
          {
            \\"error_caching_min_ttl\\": 300,
            \\"error_code\\": 404,
            \\"response_code\\": 200,
            \\"response_page_path\\": \\"/index.html\\"
          }
        ],
        \\"default_cache_behavior\\": {
          \\"allowed_methods\\": [
            \\"GET\\",
            \\"HEAD\\"
          ],
          \\"cache_policy_id\\": \\"658327ea-f89d-4fab-a63d-7e88639e58f6\\",
          \\"cached_methods\\": [
            \\"GET\\",
            \\"HEAD\\"
          ],
          \\"compress\\": true,
          \\"default_ttl\\": 3600,
          \\"field_level_encryption_id\\": \\"1234567890abcdef1234567890abcdef\\",
          \\"forwarded_values\\": {
            \\"cookies\\": {
              \\"forward\\": \\"none\\"
            },
            \\"headers\\": [
              \\"*\\"
            ],
            \\"query_string\\": false,
            \\"query_string_cache_keys\\": [
              \\"*\\"
            ]
          },
          \\"function_association\\": [
            {
              \\"event_type\\": \\"viewer-request\\",
              \\"function_arn\\": \\"arn:aws:lambda:us-east-1:111111111111:function:my-function\\"
            }
          ],
          \\"lambda_function_association\\": [
            {
              \\"event_type\\": \\"origin-request\\",
              \\"include_body\\": true,
              \\"lambda_arn\\": \\"arn:aws:lambda:us-east-1:111111111111:function:my-function\\"
            }
          ],
          \\"max_ttl\\": 86400,
          \\"min_ttl\\": 0,
          \\"origin_request_policy_id\\": \\"658327ea-f89d-4fab-a63d-7e88639e58f6\\",
          \\"realtime_log_config_arn\\": \\"arn:aws:logs:us-east-1:111111111111:realtime-log-config/my-distribution\\",
          \\"response_headers_policy_id\\": \\"658327ea-f89d-4fab-a63d-7e88639e58f6\\",
          \\"smooth_streaming\\": true,
          \\"target_origin_id\\": \\"my-target-origin\\",
          \\"trusted_key_groups\\": [
            \\"1234567890abcdef1234567890abcdef\\"
          ],
          \\"trusted_signers\\": [
            \\"1234567890abcdef1234567890abcdef\\"
          ],
          \\"viewer_protocol_policy\\": \\"redirect-to-https\\"
        },
        \\"default_root_object\\": \\"index.html\\",
        \\"enabled\\": true,
        \\"http_version\\": \\"http2\\",
        \\"is_ipv6_enabled\\": true,
        \\"logging_config\\": {
          \\"bucket\\": \\"my-logs.s3.amazonaws.com\\",
          \\"include_cookies\\": true,
          \\"prefix\\": \\"myprefix\\"
        },
        \\"ordered_cache_behavior\\": [
          {
            \\"allowed_methods\\": [
              \\"GET\\",
              \\"HEAD\\"
            ],
            \\"cache_policy_id\\": \\"658327ea-f89d-4fab-a63d-7e88639e58f6\\",
            \\"cached_methods\\": [
              \\"GET\\",
              \\"HEAD\\"
            ],
            \\"compress\\": true,
            \\"default_ttl\\": 3600,
            \\"field_level_encryption_id\\": \\"1234567890abcdef1234567890abcdef\\",
            \\"forwarded_values\\": {
              \\"cookies\\": {
                \\"forward\\": \\"none\\"
              },
              \\"headers\\": [
                \\"*\\"
              ],
              \\"query_string\\": false,
              \\"query_string_cache_keys\\": [
                \\"*\\"
              ]
            },
            \\"function_association\\": [
              {
                \\"event_type\\": \\"viewer-request\\",
                \\"function_arn\\": \\"arn:aws:lambda:us-east-1:111111111111:function:my-function\\"
              }
            ],
            \\"lambda_function_association\\": [
              {
                \\"event_type\\": \\"origin-request\\",
                \\"include_body\\": true,
                \\"lambda_arn\\": \\"arn:aws:lambda:us-east-1:111111111111:function:my-function\\"
              }
            ],
            \\"max_ttl\\": 86400,
            \\"min_ttl\\": 0,
            \\"origin_request_policy_id\\": \\"658327ea-f89d-4fab-a63d-7e88639e58f6\\",
            \\"path_pattern\\": \\"/images/*\\",
            \\"realtime_log_config_arn\\": \\"arn:aws:logs:us-east-1:111111111111:realtime-log-config/my-distribution\\",
            \\"response_headers_policy_id\\": \\"658327ea-f89d-4fab-a63d-7e88639e58f6\\",
            \\"smooth_streaming\\": true,
            \\"target_origin_id\\": \\"my-target-origin\\",
            \\"trusted_key_groups\\": [
              \\"1234567890abcdef1234567890abcdef\\"
            ],
            \\"trusted_signers\\": [
              \\"1234567890abcdef1234567890abcdef\\"
            ],
            \\"viewer_protocol_policy\\": \\"redirect-to-https\\"
          }
        ],
        \\"origin\\": [
          {
            \\"connection_attempts\\": 3,
            \\"connection_timeout\\": 10,
            \\"custom_header\\": [
              {
                \\"name\\": \\"MyCustomHeader\\",
                \\"value\\": \\"MyCustomValue\\"
              }
            ],
            \\"custom_origin_config\\": {
              \\"http_port\\": 80,
              \\"https_port\\": 443,
              \\"origin_keepalive_timeout\\": 5,
              \\"origin_protocol_policy\\": \\"https-only\\",
              \\"origin_read_timeout\\": 30,
              \\"origin_ssl_protocols\\": [
                \\"TLSv1.2\\"
              ]
            },
            \\"domain_name\\": \\"example.com\\",
            \\"origin_access_control_id\\": \\"1234567890abcdef1234567890abcdef\\",
            \\"origin_id\\": \\"my-origin\\",
            \\"origin_path\\": \\"/mypath\\",
            \\"origin_shield\\": {
              \\"enabled\\": true
            },
            \\"s3_origin_config\\": {
              \\"origin_access_identity\\": \\"origin-access-identity/cloudfront/E127EXAMPLE51Z\\"
            }
          }
        ],
        \\"origin_group\\": [
          {
            \\"failover_criteria\\": {
              \\"status_codes\\": [
                500,
                502
              ]
            },
            \\"member\\": [
              {
                \\"origin_id\\": \\"my-origin\\"
              }
            ],
            \\"origin_id\\": \\"my-origin-group\\"
          }
        ],
        \\"price_class\\": \\"PriceClass_100\\",
        \\"restrictions\\": {
          \\"geo_restriction\\": {
            \\"locations\\": [
              \\"US\\",
              \\"CA\\",
              \\"GB\\"
            ],
            \\"restriction_type\\": \\"whitelist\\"
          }
        },
        \\"staging\\": true,
        \\"tags\\": {
          \\"Name\\": \\"my-distribution\\"
        },
        \\"viewer_certificate\\": {
          \\"acm_certificate_arn\\": \\"arn:aws:acm:us-east-1:111111111111:certificate/12345678-1234-1234-1234-123456789012\\",
          \\"minimum_protocol_version\\": \\"TLSv1.2_2019\\",
          \\"ssl_support_method\\": \\"sni-only\\"
        },
        \\"web_acl_id\\": \\"arn:aws:wafv2:us-east-1:111111111111:regional/webacl/my-web-acl\\"
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_cloudfront_distribution.resource_22C949BF.domain_name, \\\\\\",\\\\\\", aws_cloudfront_distribution.resource_22C949BF.id])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;

exports[`CloudFront > Should map AWS::CloudFront::Distribution 2`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_cloudfront_distribution\\": {
      \\"resource_22C949BF\\": {
        \\"default_cache_behavior\\": {
          \\"allowed_methods\\": [
            \\"GET\\",
            \\"HEAD\\"
          ],
          \\"cache_policy_id\\": \\"658327ea-f89d-4fab-a63d-7e88639e58f6\\",
          \\"cached_methods\\": [
            \\"GET\\",
            \\"HEAD\\"
          ],
          \\"compress\\": true,
          \\"default_ttl\\": 3600,
          \\"field_level_encryption_id\\": \\"1234567890abcdef1234567890abcdef\\",
          \\"forwarded_values\\": {
            \\"cookies\\": {
              \\"forward\\": \\"none\\"
            },
            \\"headers\\": [
              \\"*\\"
            ],
            \\"query_string\\": false,
            \\"query_string_cache_keys\\": [
              \\"*\\"
            ]
          },
          \\"function_association\\": [
            {
              \\"event_type\\": \\"viewer-request\\",
              \\"function_arn\\": \\"arn:aws:lambda:us-east-1:111111111111:function:my-function\\"
            }
          ],
          \\"lambda_function_association\\": [
            {
              \\"event_type\\": \\"origin-request\\",
              \\"include_body\\": true,
              \\"lambda_arn\\": \\"arn:aws:lambda:us-east-1:111111111111:function:my-function\\"
            }
          ],
          \\"max_ttl\\": 86400,
          \\"min_ttl\\": 0,
          \\"origin_request_policy_id\\": \\"658327ea-f89d-4fab-a63d-7e88639e58f6\\",
          \\"realtime_log_config_arn\\": \\"arn:aws:logs:us-east-1:111111111111:realtime-log-config/my-distribution\\",
          \\"response_headers_policy_id\\": \\"658327ea-f89d-4fab-a63d-7e88639e58f6\\",
          \\"smooth_streaming\\": true,
          \\"target_origin_id\\": \\"my-target-origin\\",
          \\"trusted_key_groups\\": [
            \\"1234567890abcdef1234567890abcdef\\"
          ],
          \\"trusted_signers\\": [
            \\"1234567890abcdef1234567890abcdef\\"
          ],
          \\"viewer_protocol_policy\\": \\"redirect-to-https\\"
        },
        \\"enabled\\": true,
        \\"origin\\": [
          {
            \\"connection_attempts\\": 3,
            \\"connection_timeout\\": 10,
            \\"custom_header\\": [
              {
                \\"name\\": \\"MyCustomHeader\\",
                \\"value\\": \\"MyCustomValue\\"
              }
            ],
            \\"domain_name\\": \\"example.com\\",
            \\"origin_access_control_id\\": \\"1234567890abcdef1234567890abcdef\\",
            \\"origin_id\\": \\"my-origin\\",
            \\"origin_path\\": \\"/mypath\\",
            \\"origin_shield\\": {
              \\"enabled\\": true
            },
            \\"s3_origin_config\\": {
              \\"origin_access_identity\\": \\"origin-access-identity/cloudfront/E127EXAMPLE51Z\\"
            }
          }
        ],
        \\"restrictions\\": {
          \\"geo_restriction\\": {
            \\"locations\\": [
              \\"US\\",
              \\"CA\\",
              \\"GB\\"
            ],
            \\"restriction_type\\": \\"whitelist\\"
          }
        },
        \\"viewer_certificate\\": {
          \\"acm_certificate_arn\\": \\"arn:aws:acm:us-east-1:111111111111:certificate/12345678-1234-1234-1234-123456789012\\",
          \\"minimum_protocol_version\\": \\"TLSv1.2_2019\\",
          \\"ssl_support_method\\": \\"sni-only\\"
        }
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.64.0\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${join(\\\\\\"\\\\\\", [aws_cloudfront_distribution.resource_22C949BF.domain_name, \\\\\\",\\\\\\", aws_cloudfront_distribution.resource_22C949BF.id])}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;
